pkg/api/types.go

```diff
@@ -1469,8 +1469,10 @@ type NodeSystemInfo struct {
 
 // NodeStatus is information about the current status of a node.
 type NodeStatus struct {
-	// Capacity represents the available resources of a node.
+	// Capacity represents the total resources of a node.
 	Capacity ResourceList `json:"capacity,omitempty"`
+	// Allocatable represents the resources of a node that are available for scheduling.
+	Allocatable ResourceList `json:"allocatable,omitempty"`
```

Comment:
As of now our scheduler is built on this assumption. We will only add more
functionality based on this assumption.
If nodes can also vertically scale, how about making that change to the API
after ensuring that all the components can handle that change?

On Wed, Dec 16, 2015 at 5:04 PM, Tim St. Clair <notifications@github.com>
wrote:

> In pkg/api/types.go
> <https://github.com/kubernetes/kubernetes/pull/18611#discussion_r47857609>
> :
>
> >  	Capacity ResourceList `json:"capacity,omitempty"`
> > +	// Allocatable represents the resources of a node that are available for scheduling.
> > +	Allocatable ResourceList `json:"allocatable,omitempty"`
>
> I'd prefer we don't make that assumption, unless there's a specific
> requirement. I suppose we could make the weak assertion that "Changes are
> relatively infrequent". But I think that by having it in status (and not
> spec), we're suggesting that it can change.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/18611/files#r47857609>.
>